signalfx-java/src/main/java/com/signalfx/connection/RetryStrategy.java

```diff
@@ -0,0 +1,25 @@
+package com.signalfx.connection;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.protocol.HttpContext;
+
+public class RetryStrategy implements ServiceUnavailableRetryStrategy {
+    private final int maxRetries;
+
+    public RetryStrategy(final int maxRetries) {
+        this.maxRetries = maxRetries;
+    }
+
+    @Override
+    public boolean retryRequest(final HttpResponse httpResponse, final int executionCount, final HttpContext httpContext) {
+        final int statusCode = httpResponse.getStatusLine().getStatusCode();
+        return executionCount <= maxRetries && (statusCode == HttpStatus.SC_REQUEST_TIMEOUT || statusCode == HttpStatus.SC_GATEWAY_TIMEOUT || statusCode == 598 || statusCode == -1);
+    }
+
+    @Override
+    public long getRetryInterval() {
+        return 0;
```

Comment:
Does this mean no backoff and that it just retries immediately? Is this desired?

Reply:
This is the behaviour in the signalfx go client so just ported it over. We could add a backoff property here should we see fit.